src/python/pants/engine/internals/graph.py

```diff
@@ -865,13 +873,43 @@ async def resolve_dependencies(
             t.address for t in subtargets.subtargets if t.address != request.field.address
         )
 
+    # If the target has `SpecialCasedDependencies`, such as the `archive` target having
+    # `files` and `packages` fields, then we possibly include those too. We don't want to always
+    # include those dependencies because they should often be excluded from the result due to
+    # being handled elsewhere in the calling code.
+    special_cased: Tuple[Address, ...] = ()
+    if request.include_special_cased_deps:
+        wrapped_tgt = await Get(WrappedTarget, Address, request.field.address)
+        # Unlike normal, we don't use `tgt.get()` because there may be >1 subclass of
+        # SpecialCasedDependencies.
```

Comment:
Actually, could we add a TODO pointing to #10924?

Reply:
I don't think there's any TODO we need to do here, tbh. This is idiomatic Python and idiomatic usage of the Target API. We get the functionality we need in 4 lines of code, without needing to invent any new abstractions.

The comment is solely to draw attention to why we don't use `tgt.get()` like normal. 

Is there something you think we could improve with these 4 lines?

Reply:
I think it's just confusing to me why we would iterate over the target's fields *again* here instead of making that part of the original `DependenciesRequest`. I think it was a good decision to make sure each `Field` has an `Address`, but it feels like that's useful for when we *discover* new things we need to introspect about the original target. I'm not sure it's really a *problem*, but it feels like that might be part of why the graph cycle is happening here?

Reply:
I'm not really following the argument here. At this point, we had only discovered that there is a `Dependencies` field (or subclass). We need to check if there are any other special cased dependencies fields that are in play, and the way to do that is to iterate over all the fields to see if they subclass this marker.

Reply:
> At this point, we had only discovered that there is a Dependencies field (or subclass).

What if we included the special-cased dependencies fields in the original `DependenciesRequest`?